domain/externalcontroller/models.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package externalcontroller
+
+import "github.com/juju/names/v4"
+
+// MigrationControllerInfo holds the details required to connect to a controller.
+type MigrationControllerInfo struct {
```

Comment:
This type doesn't make sense, because `ModelUUIDs` can never be handled with a `db` tag.

I would do one of 2 things:
1. Add `ModelUUIDs` to the core `ControllerInfo` type. Treat a nil slice as unspecified and ignore it. If it's populated we consider it and remove the model slice argument from `UpdateExternalController`.
2. Use `ExternalControllerArgs` from the `description` package.

I would tend towards 1, because it is a proper encapsulation of the domain concerns.

Reply:
I ended up doing 1, see 00b4929935c3ca2cbe55f54ec0315a28ea71f19a